pandas/core/frame.py

```diff
@@ -2734,30 +2734,32 @@ def duplicated(self, subset=None, take_last=False):
         -------
         duplicated : Series
         """
-        # kludge for #1833
-        def _m8_to_i8(x):
-            if issubclass(x.dtype.type, np.datetime64):
-                return x.view(np.int64)
-            return x
+        from pandas.core.groupby import get_group_index
+        from pandas.hashtable import duplicated_int64, _SIZE_HINT_LIMIT
+
+        size_hint = min(len(self), _SIZE_HINT_LIMIT)
+
+        def factorize(vals):
+            (hash_klass, vec_klass), vals = \
+                    algos._get_data_algo(vals, algos._hashtables)
+
+            uniques, table = vec_klass(), hash_klass(size_hint)
```

Comment:
@behzadnouri I think what @shoyer means as you are using the *private* cython impl of indexes. This is currently *only* used in ``algos``. and does not need to be exposed to general reader of ``frame.py``. So I would make a helper private function in ``algos.py`` that does these 3 lines (that return the labels/uniques).